### PR TITLE
Feat/support focus shortcuts autofocus addonBefore addonAfter and expandSuggestionsContainer

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**Description** :
- Support focusShortcuts prop to focus input field using custom keyboard shortcuts.
- support autoFocus prop for the input search field.
- Support addonBefore & addonAfter (props/ slots) to render custom jsx/ HTML markup before & after the input search field.
- Support expandSuggestionsContainer prop to expand/ limit the suggestions dropdown container width to the whole input wrapper/ input field.

**Libraries Affected:**

-  ReactiveSearch - web : DataSearch & CategorySearch
-  ReactiveSearch - Vue : DataSearch 

[Notion Link](https://www.notion.so/appbase/Support-searchbox-focus-with-keyboard-shortcut-3010b391595d4d0495a65e99f9591f9a)

[Docs PR Link](https://github.com/appbaseio/Docs/pull/193)

[Loom Demo: ReactiveSearch - **_web_**-  focusShortcuts, autoFocus, addonBefore, addonAfter, expandSuggestionsContainer](https://www.loom.com/share/1d77603df8a440708d7900af1312a82d)

[Loom Demo: ReactiveSearch - **_Vue_**-  focusShortcuts, autoFocus, addonBefore, addonAfter, expandSuggestionsContainer](https://www.loom.com/share/0861e477deff4b94a7928b82d384a837)

### CODE REVIEW

File: packages/web/src/types.ts [hunk 1]
```diff
@@ -110,6 +110,8 @@ export type rangeLabelsAlign = 'left' | 'right';
 
 export type title = string | any;
 
+export type focusShortcuts = any[];
```
Comment (bietkul) [inline packages/web/src/types.ts]: @mohdashraf010897 Can you use the property type? I think it should be an array of strings or numbers.

File: packages/web/package.json [hunk 2]
```diff
@@ -50,7 +50,8 @@
 	},
 	"peerDependencies": {
 		"react": ">=15",
-		"react-dom": ">=15"
+		"react-dom": ">=15",
+		"hotkeys-js": "^3.8.5"
```
Comment (bietkul) [inline packages/web/package.json:54]: @mohdashraf010897 Share a demo when `focusShortcuts` prop has not been used. It shouldn't require the `hotkeys-js`.

File: packages/vue/src/components/search/DataSearch.jsx [hunk 3]
```diff
@@ -504,12 +550,12 @@ const DataSearch = {
 		},
 		handleVoiceResults({ results }) {
 			if (
-				results
-				&& results[0]
-				&& results[0].isFinal
-				&& results[0][0]
-				&& results[0][0].transcript
-				&& results[0][0].transcript.trim()
+				results &&
```
Comment (bietkul) [inline packages/vue/src/components/search/DataSearch.jsx]: @mohdashraf010897 Why formatting changes? Are you using some different prettier config?

**Review by bietkul — CHANGES_REQUESTED** [review]
@mohdashraf010897 Please fix formatting changes. There a lot of unnecessary changes because of some prettier issues.

File: packages/web/package.json (hunk 2 above)
Comment (mohdashraf010897) [reply packages/web/package.json:54]: https://www.loom.com/share/878a4f0a63dd43c99f8169ec52fcd87a

**Review by bietkul — APPROVED** [review]

**Comment by bietkul** [comment]
@mohdashraf010897 You should create stories for search components in `react` and `vue` playgrounds.
- with `autoFocus`
- with `focusShortcuts`
- with `addonBefore`
- with `addonAfter`
- with `addonBefore` and `addonAfter`
- with `expandSuggestionsContainer`

**Comment by mohdashraf010897** [comment]
> @mohdashraf010897 You should create stories for search components in `react` and `vue` playgrounds.
> 
> * with `autoFocus`
> * with `focusShortcuts`
> * with `addonBefore`
> * with `addonAfter`
> * with `addonBefore` and `addonAfter`
> * with `expandSuggestionsContainer`

https://github.com/appbaseio/playground/pull/69
https://github.com/appbaseio/vue-playground/pull/32

**Review by bietkul — APPROVED** [review]

**Pull request merged**